framework/src/controller/schema/application.js

```diff
@@ -13,6 +13,211 @@ module.exports = {
 	constants: {
 		id: '#/app/constants',
 		type: 'object',
+		required: [
+			'ACTIVE_DELEGATES',
+			'BLOCK_SLOT_WINDOW',
+			'ADDITIONAL_DATA',
+			'BLOCK_RECEIPT_TIMEOUT',
+			'EPOCH_TIME',
+			'FEES',
+			'MAX_PAYLOAD_LENGTH',
+			'MAX_PEERS',
+			'MAX_SHARED_TRANSACTIONS',
+			'MAX_TRANSACTIONS_PER_BLOCK',
+			'MAX_VOTES_PER_TRANSACTION',
+			'MAX_VOTES_PER_ACCOUNT',
+			'MIN_BROADHASH_CONSENSUS',
+			'MULTISIG_CONSTRAINTS',
+			'NETHASHES',
+			'NORMALIZER',
+			'REWARDS',
+			'TOTAL_AMOUNT',
+			'UNCONFIRMED_TRANSACTION_TIMEOUT',
+			'EXPIRY_INTERVAL',
+		],
+		properties: {
+			ACTIVE_DELEGATES: {
+				type: 'integer',
+				min: 0,
```

Comment:
I think this must not be zero in any case. There would be a minimum value for it @MaciejBaj @4miners Please confirm. 

I am not sure if that's the one or `MIN_BROADHASH_CONSENSUS`, but any of these two must be minimum 51.

Reply:
There are the conditions which definitely will not work with ACTIVE_DELEGATES equal to 0, like:
```
// Check if we are one block before the last block of a round, if yes - perform round snapshot
				if ((block.height + 1) % ACTIVE_DELEGATES === 0) {
```

The value here should be min 1 and should always be odd. That's what is missing currently. 

While setting new parameters range each of required constants what should be inspected before is if the change of the parameter is affecting the application behaviour. 

@michielmulders please perform the experiments and see if the application changes the behaviour as expected after change of the most significant parameters:
1. ACTIVE_DELEGATES
2. BLOCK_SLOT_WINDOW
3. BLOCK_RECEIPT_TIMEOUT
4. EPOCH_TIME
5. MAX_PAYLOAD_LENGTH
6. MAX_TRANSACTIONS_PER_BLOCK
7. NORMALIZER

Reply:
JSON Schema does not support detection for odd/even numbers.
For strings, we can use this regex that detects odd numbers: `^\d*[13579]$`